### PR TITLE
Deploy review apps

### PR DESCRIPTION
**Pull request by NickGraham101**
### Context

We need to be able to deploy review apps to test changes.

### Changes proposed in this pull request

- Adds option to deploy a review app by adding the deploy label
- Adds a workflow to check for review app when any PR is closed and delete it

### Guidance to review

A review app has been deployed using this PR, check logs and functionality.

### Link to Trello card

https://trello.com/c/eT6yaaHD

### Checklist

- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running manually

### CODE REVIEW

**Comment by NickGraham101** [comment]
Testing delete workflow

**Comment by NickGraham101** [comment]
Reopening after successful test of delete workflow

**Comment by NickGraham101** [comment]
[Infrastructure and domain naming](https://trello.com/c/eT6yaaHD/694-infrastructure-and-domain-naming)

**Comment by NickGraham101** [comment]
Closing to retest fresh deployment

**Comment by NickGraham101** [comment]
Reopening to retest fresh deployment

File: .github/actions/deploy-environment/action.yml [hunk 1]
```diff
@@ -115,10 +130,19 @@ runs:
       shell: bash
 
     - uses: azure/webapps-deploy@v2
+      if: ${{ inputs.environment_name }} != "review"
+      with:
+        app-name: ${{ env.resource_prefix }}-${{ inputs.environment_name}}${{ env.review_app_suffix }}-app
```
Comment (RMcVelia) [inline .github/actions/deploy-environment/action.yml]: we shouldn't have env.review_app_suffix here?

File: terraform/app.tf [hunk 2]
```diff
@@ -4,6 +4,8 @@ locals {
       DOCKER_REGISTRY_SERVER_URL            = "https://ghcr.io",
       DATABASE_URL                          = "postgres://postgres@${local.postgres_server_name}.postgres.database.azure.com:5432"
       DATABASE_PASSWORD                     = local.infrastructure_secrets.POSTGRES_ADMIN_PASSWORD
+      HOSTING_DOMAIN                        = var.domain != null ? "https://${var.domain}" : "https://${local.aytq_web_app_name}.azurewebsites.net"
```
Comment (RMcVelia) [inline terraform/app.tf:7]: where are we setting var.domain?

Comment (nickdevdfe) [reply terraform/app.tf:7]: It's being set [here](https://github.com/DFE-Digital/access-your-teaching-qualifications/blob/5af6a2d00a3621b12d6e135a781e91a11ef35ea1/terraform/variables.tf#L124), but not in use yet

**Review by RMcVelia — APPROVED** [review]

**Pull request merged**